opentelemetry/proto/profiles/v1/alternatives/denormalized/denormalized.proto

```diff
@@ -0,0 +1,131 @@
+import "opentelemetry/proto/resource/v1/resource.proto";
+
+option go_package = "go.opentelemetry.io/proto/otlp/profiles/v1/alternatives/denormalized";
+
+
+// A pointer from a profile to a trace span.
+message Link {
+  // A unique identifier of a trace that this linked span is part of. The ID is a
+  // 16-byte array.
+  bytes trace_id = 1;
+  // A unique identifier for the linked span. The ID is an 8-byte array.
+  bytes span_id = 2;
+}
+
+message Sample {
+  repeated Location locations = 1;
+  repeated Link links = 2;
+  repeated opentelemetry.proto.common.v1.KeyValue attributes = 3;
+  uint32 dropped_attributes_count = 4;
+
+  // this one is repeated because there can be multiple profile kinds in one profile. Typical example is memory profiles in go that contain:
+  //   * alloc_objects
+  //   * alloc_bytes
+  //   * inuse_objects
+  //   * inuse_bytes
+  repeated fixed64 values = 5;
+
+  // optional
+  fixed64 timestamp_unix_nano = 6;
+}
+
+// borrowed from metrics proto
+enum AggregationTemporality {
+  AGGREGATION_TEMPORALITY_UNSPECIFIED = 0;
+  AGGREGATION_TEMPORALITY_DELTA = 1;
+  AGGREGATION_TEMPORALITY_CUMULATIVE = 2;
+}
+
+message SampleType {
+  AggregationTemporality aggregation_temporality = 1;
+  uint64 sample_size = 2;
+
+  // CPU / memory /etc
+  string type = 3;
+  string unit = 4;
+}
+
+message Profile {
+  repeated SampleType sample_types = 1;
+  repeated Sample samples = 2;
+}
+
+// borrowed from pprof proto
+message Mapping {
+  // Address at which the binary (or DLL) is loaded into memory.
+  uint64 memory_start = 1;
+  // The limit of the address range occupied by this mapping.
+  uint64 memory_limit = 2;
+  // Offset in the binary that corresponds to the first mapped address.
+  uint64 file_offset = 3;
+  // The object this entry is loaded from.  This can be a filename on
+  // disk for the main binary and shared libraries, or virtual
+  // abstractions like "[vdso]".
+  string filename = 4;
+  // A string that uniquely identifies a particular program version
+  // with high probability. E.g., for binaries generated by GNU tools,
+  // it could be the contents of the .note.gnu.build-id field.
+  string build_id = 5;
+
+  // The following fields indicate the resolution of symbolic info.
+  bool has_functions = 6;
+  bool has_filenames = 7;
+  bool has_line_numbers = 8;
+  bool has_inline_frames = 9;
+}
+
+// borrowed from pprof proto
+// Describes function and line table debug information.
+message Location {
+  // profile.Mapping for this location.
+  // It can be unset if the mapping is unknown or not applicable for
+  // this profile type.
+  Mapping mapping = 1;
+
+  // The instruction address for this location, if available.  It
+  // should be within [Mapping.memory_start...Mapping.memory_limit]
+  // for the corresponding mapping. A non-leaf address may be in the
+  // middle of a call instruction. It is up to display tools to find
+  // the beginning of the instruction if necessary.
+  uint64 address = 2;
```

Comment:
We need to specify *exactly* how deep into the call instruction the address is allowed to be, display tools do not have access to the assembly bytes and hence cannot do more than simple arithmetic to find the beginning of the instruction.